core/src/main/java/com/scalar/db/storage/dynamo/QueryScanner.java

```diff
@@ -49,18 +58,23 @@ private boolean hasNext() {
       return true;
     }
     if (lastEvaluatedKey != null) {
-      handleResponse(request.execute(lastEvaluatedKey));
+      if (remainingLimit != null) {
+        handleResponse(request.execute(lastEvaluatedKey, remainingLimit));
+      } else {
+        handleResponse(request.execute(lastEvaluatedKey));
+      }
```

Comment:
If we have the renaming limit, specify the limit and send a request to DynamoDB.